desc/efit.py

```diff
@@ -0,0 +1,373 @@
+
+        eq = Equilibrium(**inputs)
+
+        return eq
+
+    @classmethod
+    def efit_to_desc_input(  # noqa: C901 - fxn too complex
+        cls,
+        eq,
+        outfile,
+        objective="force",
+        optimizer="lsq-exact",
+        ftol=1e-2,
+        xtol=1e-2,
+        gtol=1e-6,
+        maxiter=100,
+        threshold=1e-8,
+    ):
+        """Generate a DESC input file from a DESC equilibrium object.
+
+        DESC will automatically choose continuation parameters
+
+        Parameters
+        ----------
+        eq: Equilibrium
+            DESC equilibrium object
+        outfile : str or path-like
+            name of the DESC input file to create
+        objective : str
+            objective type used in the input file
+        optimizer : str
+            type of optimizer
+        ftol : float
+            relative tolerance of the objective function f
+        xtol : float
+            relative tolerance of the state vector x
+        gtol : float
+            absolute tolerance of the projected gradient g
+        maxiter : int
+            maximum number of optimizer iterations per continuation step
+        threshold : float
+            Fourier coefficients below this value will be set to 0.
+        """
+        with open(outfile, "w+") as f:
+            header = "# DESC input file generated from an geqdsk file\n# "
+            f.write(header + "\n")
+
+            f.write("\n# global parameters\n")
+            f.write("sym = {:d}\n".format(eq.sym))
+            f.write("NFP = {:d}\n".format(int(eq.NFP)))
+            f.write("Psi = {:.8E}\n".format(eq.Psi))
+
+            f.write("\n# spectral resolution\n")
+            for key, val in {
+                "L_rad": "L",
+                "M_pol": "M",
+                "N_tor": "N",
+                "L_grid": "L_grid",
+                "M_grid": "M_grid",
+                "N_grid": "N_grid",
+            }.items():
+                f.write(f"{key} = {getattr(eq, val)}\n")
+
+            f.write("\n# solver tolerances\n")
+            f.write("ftol = {:.3E}\n".format(ftol))
+            f.write("xtol = {:.3E}\n".format(xtol))
+            f.write("gtol = {:.3E}\n".format(gtol))
+            f.write("maxiter = {:d}\n".format(maxiter))
+
+            f.write("\n# solver methods\n")
+            f.write(f"optimizer = {optimizer}\n")
+            f.write(f"objective = {objective}\n")
+            f.write("spectral_indexing = {}\n".format(eq._spectral_indexing))
+
+            f.write("\n# pressure and rotational transform/current profiles\n")
+
+            grid = LinearGrid(L=eq.L_grid, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+            rho = grid.nodes[grid._unique_rho_idx, 0]
+
+            pressure = grid.compress(eq.compute("p", grid=grid)["p"])
+            iota = grid.compress(eq.compute("iota", grid=grid)["iota"])
+
+            pres_profile = PowerSeriesProfile.from_values(
+                rho, pressure, order=eq.L, sym=False
```

Comment:
we probably want to enforce sym=True? since physically it should be an even function of rho? in the input file you have, the coeffs for the odd parts are super small so maybe it does not matter but things will throw warnings with sym=False